src/apollo/typedef/schema.graphql

```diff
@@ -0,0 +1,81 @@
+scalar Url
+
+enum Role {
+  APPLICANT
+  EMPLOYER
+  ADMIN
+}
+
+enum ApplicationStatus {
+  SUBMITTED
+  REVIEWING
+  REJECTED
+  ACCEPTED
+}
+
+interface User {
+  id: ID!
+  email: String!
+  password: String!
+  role: Role!
+}
+
+type Admin {
+  id: ID!
+  email: String!
+  password: String!
+  role: Role!
+}
+
+type Applicant implements User {
```

Comment:
Oooo first time seeing implements in typescript!

Reply:
Oh, this is for the graphql schema. For graphql models, `implements` is somewhat like `extends` in ts.

I'm not sure if you've looked into `implements` for ts at the time of me writing this comment. But yeah, for ts, `implements` is when you have a interface that you want to adhere with another interface, strictly speaking.